pkgs/development/python-modules/canmatrix/default.nix

```diff
@@ -1,65 +1,99 @@
 { lib
-, buildPythonPackage
-, fetchFromGitHub
-, pythonOlder
 , attrs
-, bitstruct
+, buildPythonPackage
 , click
+, fetchFromGitHub
 , future
-, pathlib2
-, typing ? null
+, importlib-metadata
+, ldfparser
 , lxml
-, xlwt
-, xlrd
-, xlsxwriter
-, pyyaml
+, openpyxl
 , pytestCheckHook
+, pythonOlder
+, pyyaml
+, six
+, xlrd
+, xlwt
 }:
 
 buildPythonPackage rec {
   pname = "canmatrix";
   version = "1.0";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.7";
 
-  # uses fetchFromGitHub as PyPi release misses test/ dir
   src = fetchFromGitHub {
     owner = "ebroecker";
-    repo = pname;
+    repo = "canmatrix";
     rev = "refs/tags/${version}";
-    sha256 = "sha256-UUJnLVt+uOj8Eav162btprkUeTemItGrSnBBB9UhJJI=";
+    hash = "sha256-UUJnLVt+uOj8Eav162btprkUeTemItGrSnBBB9UhJJI=";
   };
 
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace "version = versioneer.get_version()" 'version = "${version}"'
+  '';
+
   propagatedBuildInputs = [
-    # required
     attrs
-    bitstruct
     click
     future
-    pathlib2
-    # optional
-    lxml
-    xlwt
-    xlrd
-    xlsxwriter
-    pyyaml
-  ] ++ lib.optional (pythonOlder "3.5") typing;
+    six
+  ] ++ lib.optionals (pythonOlder "3.8") [
+    importlib-metadata
+  ];
 
-  postPatch = ''
-    substituteInPlace setup.py \
-      --replace "version = versioneer.get_version()" "version = \"${version}\""
-  '';
+  passthru.optional-dependencies = {
+    arxml = [
+      lxml
+    ];
+    fibex = [
+      lxml
+    ];
+    kcd = [
+      lxml
+    ];
+    ldf = [
+      ldfparser
+    ];
+    odx = [
+      lxml
+    ];
+    xls = [
+      xlrd
+      xlwt
+    ];
+    xlsx = [
+      openpyxl
+    ];
+    yaml = [
+      pyyaml
+    ];
+  };
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ] ++ lib.flatten (builtins.attrValues passthru.optional-dependencies);
 
-  nativeCheckInputs = [ pytestCheckHook ];
-  # long_envvar_name_imports requires stable key value pair ordering
-  pytestFlagsArray = [ "-s src/canmatrix" ];
-  disabledTests = [ "long_envvar_name_imports" ];
-  pythonImportsCheck = [ "canmatrix" ];
+  pytestFlagsArray = [
+    # long_envvar_name_imports requires stable key value pair ordering
+    "-s src/canmatrix"
```

Comment:
@fabaff Is this comment for `"-s"` or for `"src/canmatrix"`?

Sorry for bothering. I'm currently migrating `pytestFlagsArray` to `pytestFlags` and need to separate the two command-line arguments, hence the question.